timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java

```diff
@@ -111,47 +114,79 @@
             .around(TIMELOCK_SERVER_HOLDER);
 
     @Test
-    public void notExceedingThreadCountLimitsSucceeds() throws Exception {
+    public void singleClientCanUseLocalAndSharedThreads() throws Exception {
         List<RemoteLockService> lockService = ImmutableList.of(getLockService(CLIENT_1));
 
         assertThat(lockAndUnlockAndCountExceptions(lockService, LOCAL_TC_LIMIT + SHARED_TC_LIMIT))
                 .isEqualTo(0);
     }
 
     @Test
-    public void multipleClientsCanShareGlobalThreads() throws Exception {
+    public void multipleClientsCanUseSharedThreads() throws Exception {
         List<RemoteLockService> lockServiceList = ImmutableList.of(
                 getLockService(CLIENT_1), getLockService(CLIENT_2), getLockService(CLIENT_3));
 
         assertThat(lockAndUnlockAndCountExceptions(lockServiceList, LOCAL_TC_LIMIT + SHARED_TC_LIMIT / 3))
                 .isEqualTo(0);
     }
 
-    private int lockAndUnlockAndCountExceptions(List<RemoteLockService> lockServices, int numRequests)
-            throws Exception {
-        CountDownLatch countDownLatch = new CountDownLatch(lockServices.size() * numRequests);
-        ExecutorService executorService = Executors.newFixedThreadPool(lockServices.size() * numRequests);
-        Map<RemoteLockService, List<Future<LockRefreshToken>>> futureMap = new HashMap<>();
-        lockServices.forEach(
-                service -> futureMap.put(service, requestLocks(service, numRequests, executorService, countDownLatch)));
-        return futureMap.entrySet().stream()
-                .mapToInt(entry -> unlockAndCountExceptions(entry.getKey(), entry.getValue())).sum();
+    @Test
+    public void throwOnSingleClientRequestingSameLock() throws Exception {
+        List<RemoteLockService> lockServiceList = ImmutableList.of(
+                getLockService(CLIENT_1));
+        int exceedingRequests = 10;
+        int maxRequestsForOneClient = LOCAL_TC_LIMIT + SHARED_TC_LIMIT;
+
+        assertThat(lockAndUnlockAndCountExceptions(lockServiceList, exceedingRequests + maxRequestsForOneClient))
+                .isEqualTo(exceedingRequests);
+    }
+
+    @Test
+    public void throwOnTwoClientsRequestingSameLock() throws Exception {
+        List<RemoteLockService> lockServiceList = ImmutableList.of(
+                getLockService(CLIENT_1), getLockService(CLIENT_2));
+        int requestsPerClient = 80;
+        int maxRequestsForTwoClients = 2 * LOCAL_TC_LIMIT + SHARED_TC_LIMIT;
+
+        assertThat(lockAndUnlockAndCountExceptions(lockServiceList, requestsPerClient))
+                .isEqualTo(2 * requestsPerClient - maxRequestsForTwoClients);
```

Comment:
That's smart. Your first suggestion has an off-by-one error (SHARED_TC_LIMIT + exceedingRequests is odd), but the second should work. Thanks!